source/server/server_configuration_parser.cpp

```diff
@@ -141,6 +142,21 @@ int ServerConfigurationParser::parse_max_message_size() const
   return UNLIMITED_MAX_MESSAGE_SIZE;
 }
 
+int ServerConfigurationParser::parse_sideband_port() const
+{
+  auto sideband_port = config_file_.find(kSidebandPortJsonKey);
+  if (sideband_port != config_file_.end()) {
+    try {
+      return sideband_port->get<int>();
+    }
+    catch (const nlohmann::json::type_error& ex) {
+      throw InvalidMaxMessageSizeException(ex.what());
```

Comment:
This should probably throw a `WrongPortTypeException(ex.what());` instead.

Also, consider checking the port range like `parse_port` on line 271 below, and if it's not in a valid range throwing an `InvalidPortException`.

Reply:
I've created a helper which does the port extraction.
I'm now using that at two places to get the port.